code/game/objects/items/devices/motion_detector.dm

```diff
@@ -86,7 +86,7 @@
 		human_user = loc
 
 	var/detected
-	for(var/mob/M in living_mob_list)
+	for(var/mob/living/M in range(detector_range, src))
```

Comment:
I dont think this is more performant, this gets a larger list than the one above most likely and has to do an expensive operation to get said list

Reply:
I can't possibly see this being true except in extreme low pop.

Reply:
if it was living_players then it would definitely be true, you'd be surprised how many objects you can find in a 5x5 range of something.

Reply:
Apparently MSO did a performance test of this fix.

Reply:
```
/obj/item/device/motiondetector/proc/process_old                                                 0.011        0.089        0.092          375
/obj/item/device/motiondetector/proc/process_new                                                 0.004        0.054        0.054          381
```

Reply:
https://i.gyazo.com/7a08ea4db2822acb2a042f0037e4072e.png
It does depend on the number of objects near.